docs/reference/eql/functions.asciidoc

```diff
@@ -805,6 +806,84 @@ If using a field as the argument, this parameter supports only
 *Returns:* integer, float, or `null`
 ====
 
+[discrete]
+[[eql-fn-number]]
+=== `number`
+
+Converts a string to the corresponding integer or float.
+
+[%collapsible]
+====
+*Example*
+[source,eql]
+----
+number("1337")              // returns 1337
+number("42.5")              // returns 42.5
+number("0xdeadbeef", 16)    // returns 3735928559
```

Comment:
The `0x` prefix will be stripped off, so these are both supported and equivalent
```suggestion
number("deadbeef", 16)      // returns 3735928559
number("0xdeadbeef", 16)    // returns 3735928559
```

Also, if `0x` is provided and there is no base, it will be autodetected as hex
```c
number("0xdeadbeef")    // returns 3735928559
```